<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Sequence logos for DNA sequence alignments • seqLogo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Sequence logos for DNA sequence alignments">
<meta property="og:description" content="seqLogo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">seqLogo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.57.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/seqLogo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ivanek/seqLogo/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="seqLogo_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Sequence logos for DNA sequence alignments</h1>
                        <h4 class="author">Oliver Bembom</h4>
            <address class="author_afil">
      Division of Biostatistics, University of California, Berkeley<br><h4 class="author">Robert Ivánek</h4>
            <address class="author_afil">
      Department of Biomedicine, University of Basel, Basel, Switzerland<br><a class="author_email" href="mailto:#"></a><a href="mailto:robert.ivanek@unibas.ch" class="email">robert.ivanek@unibas.ch</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/ivanek/seqLogo/blob/master/vignettes/seqLogo.Rmd"><code>vignettes/seqLogo.Rmd</code></a></small>
      <div class="hidden name"><code>seqLogo.Rmd</code></div>

    </address>
</div>

    
    
<p><strong>Last edited</strong>: 08 May 2020</p>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>An alignment of DNA or amino acid sequences is commonly represented in the form of a position weight matrix (PWM), a <span class="math inline">\(J \times W\)</span> matrix in which position <span class="math inline">\((j,w)\)</span> gives the probability of observing nucleotide <span class="math inline">\(j\)</span> in position <span class="math inline">\(w\)</span> of an alignment of length <span class="math inline">\(W\)</span>. Here <span class="math inline">\(J\)</span> denotes the number of letters in the alphabet from which the sequences were derived. An important summary measure of a given position weight matrix is its information content profile <span class="citation">(Schneider et al. 1986)</span>. The information content at position <span class="math inline">\(w\)</span> of the motif is given by</p>
<p><span class="math display">\[
IC(w) = \log_2(J) + \sum_{j=1}^J p_{wj}\log_2(p_{wj}) = \log_2(J) - entropy(w).
\]</span></p>
<p>The information content is measured in bits and, in the case of DNA sequences, ranges from 0 to 2 bits. A position in the motif at which all nucleotides occur with equal probability has an information content of 0 bits, while a position at which only a single nucleotide can occur has an information content of 2 bits. The information content at a given position can therefore be thought of as giving a measure of the tolerance for substitutions in that position: Positions that are highly conserved and thus have a low tolerance for substitutions correspond to high information content, while positions with a high tolerance for substitutions correspond to low information content.</p>
<p>Sequence logos are a graphical representation of sequence alignments developed by <span class="citation">(Schneider and Stephens 1990)</span>. Each logo consists of stacks of symbols, one stack for each position in the sequence. The overall height of the stack is proportional to the information content at that position, while the height of symbols within the stack indicates the relative frequency of each amino or nucleic acid at that position. In general, a sequence logo provides a richer and more precise description of, for example, a binding site, than would a consensus sequence.</p>
</div>
<div id="software-implementation" class="section level1">
<h1 class="hasAnchor">
<a href="#software-implementation" class="anchor"></a>Software implementation</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.13/seqLogo">seqLogo</a></em> package provides an R implementation for plotting such sequence logos for alignments consisting of DNA sequences. Before being able to access this functionality, the user is required to load the package using the <code><a href="https://rdrr.io/r/base/library.html">library()</a></code> command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">seqLogo</span><span class="op">)</span> </code></pre></div>
<div id="the-pwm-class" class="section level2">
<h2 class="hasAnchor">
<a href="#the-pwm-class" class="anchor"></a>The <code>pwm-class</code>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.13/seqLogo">seqLogo</a></em> package defines the class <code>pwm</code> which can be used to represent position weight matrices. An instance of this class can be constructed from a simple matrix or a data frame using the function <code><a href="../reference/makePWM.html">makePWM()</a></code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/pwm1"</span>, package<span class="op">=</span><span class="st">"seqLogo"</span><span class="op">)</span>
<span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">mFile</span><span class="op">)</span>
<span class="va">m</span>
<span class="co">##    V1  V2  V3  V4  V5  V6  V7  V8</span>
<span class="co">## 1 0.0 0.0 0.0 0.3 0.2 0.0 0.0 0.0</span>
<span class="co">## 2 0.8 0.2 0.8 0.3 0.4 0.2 0.8 0.2</span>
<span class="co">## 3 0.2 0.8 0.2 0.4 0.3 0.8 0.2 0.8</span>
<span class="co">## 4 0.0 0.0 0.0 0.0 0.1 0.0 0.0 0.0</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makePWM.html">makePWM</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></code></pre></div>
<p><code><a href="../reference/makePWM.html">makePWM()</a></code> checks that all column probabilities add up to 1.0 and also obtains the information content profile and consensus sequence for the position weight matrix. These can then be accessed through the corresponding slots of the created object:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">slotNames</span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">## [1] "pwm"       "width"     "ic"        "alphabet"  "consensus"</span>
<span class="fu"><a href="../reference/pwm-class.html">pwm</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">##     1   2   3   4   5   6   7   8</span>
<span class="co">## A 0.0 0.0 0.0 0.3 0.2 0.0 0.0 0.0</span>
<span class="co">## C 0.8 0.2 0.8 0.3 0.4 0.2 0.8 0.2</span>
<span class="co">## G 0.2 0.8 0.2 0.4 0.3 0.8 0.2 0.8</span>
<span class="co">## T 0.0 0.0 0.0 0.0 0.1 0.0 0.0 0.0</span>
<span class="fu"><a href="../reference/pwm-class.html">ic</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">## [1] 1.2780719 1.2780719 1.2780719 0.4290494 0.1535607 1.2780719 1.2780719</span>
<span class="co">## [8] 1.2780719</span>
<span class="fu"><a href="../reference/pwm-class.html">consensus</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="co">## [1] "CGCGCGCG"</span></code></pre></div>
</div>
<div id="plotting-sequence-logos" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-sequence-logos" class="anchor"></a>Plotting sequence logos</h2>
<p>The <code><a href="../reference/seqLogo.html">seqLogo()</a></code> function plots sequence logos.</p>
<div id="input" class="section level3">
<h3 class="hasAnchor">
<a href="#input" class="anchor"></a>Input</h3>
<ol style="list-style-type: decimal">
<li><p>The position weight matrix for which the sequence logo is to be plotted, <code>pwm</code>. This may be either an instance of class <code>pwm</code>, as defined by the package <em><a href="https://bioconductor.org/packages/3.13/seqLogo">seqLogo</a></em>, a <code>matrix</code>, or a <code>data.frame</code>.</p></li>
<li><p>A <code>logical</code> <code>ic.scale</code> indicating whether the height of each column is to be proportional to its information content, as originally proposed by <span class="citation">(Schneider et al. 1986)</span>. If <code>ic.scale=FALSE</code>, all columns have the same height.</p></li>
</ol>
</div>
<div id="example" class="section level3">
<h3 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h3>
<p>The call <code><a href="../reference/seqLogo.html">seqLogo(p)</a></code> produces the sequence logo shown in figure @ref(seqlogo1). Alternatively, we can use <code><a href="../reference/seqLogo.html">seqLogo(p, ic.scale=FALSE)</a></code> to obtain the sequence logo shown in figure @ref(seqlogo2) in which all columns have the same height.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/seqLogo.html">seqLogo</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="seqLogo_files/figure-html/seqLogo1-1.png" alt="Sequence logo with column heights proportional to information content." width="576"><p class="caption">
Sequence logo with column heights proportional to information content.
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/seqLogo.html">seqLogo</a></span><span class="op">(</span><span class="va">p</span>, ic.scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="seqLogo_files/figure-html/seqLogo2-1.png" alt="Sequence logo with uniform column heights." width="576"><p class="caption">
Sequence logo with uniform column heights.
</p>
</div>
<p>It is also possible to change the default colors by providing a named character vector as a <code>fill</code> argument <code>seqLogo</code> function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/seqLogo.html">seqLogo</a></span><span class="op">(</span><span class="va">p</span>, fill<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="st">"#4daf4a"</span>, C<span class="op">=</span><span class="st">"#377eb8"</span>, G<span class="op">=</span><span class="st">"#ffd92f"</span>, T<span class="op">=</span><span class="st">"#e41a1c"</span><span class="op">)</span>,
        ic.scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="seqLogo_files/figure-html/seqLogo3-1.png" alt="Sequence logo with user specified colors." width="576"><p class="caption">
Sequence logo with user specified colors.
</p>
</div>
<p>The RNA logos are supported as well. In this particular case, the <code>seqLogo</code> will either accept <code>fill</code> colors specified for <code><a href="https://rdrr.io/r/base/c.html">c("A", "C", "G", "U")</a></code> letters or <code><a href="https://rdrr.io/r/base/c.html">c("A", "C", "G", "T")</a></code> and uses the color specified in element “T” for letter “U”.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makePWM.html">makePWM</a></span><span class="op">(</span><span class="va">m</span>, alphabet<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span>
<span class="fu"><a href="../reference/seqLogo.html">seqLogo</a></span><span class="op">(</span><span class="va">r</span>, ic.scale<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="seqLogo_files/figure-html/seqLogo4-1.png" alt="RNA Sequence logo." width="576"><p class="caption">
RNA Sequence logo.
</p>
</div>
</div>
</div>
<div id="software-design" class="section level2">
<h2 class="hasAnchor">
<a href="#software-design" class="anchor"></a>Software Design</h2>
<p>The following features of the programming approach employed in <em><a href="https://bioconductor.org/packages/3.13/seqLogo">seqLogo</a></em> may be of interest to users.</p>
<p><strong>Class/method object-oriented programming</strong>. Like many other Bioconductor packages, <em><a href="https://bioconductor.org/packages/3.13/seqLogo">seqLogo</a></em> has adopted the <em>S4 class/method objected-oriented programming approach</em> presented in <span class="citation">(Chambers 1998)</span>. In particular, a new class, <code>pwm</code>, is defined to represent a position weight matrix. The plot method for this class is set to produce the sequence logo corresponding to this class.</p>
<p><strong>Use of the <code>grid</code> package</strong>. The <code>grid</code> package is used to draw the sequence letters from graphical primitives. We note that this should make it easy to extend the package to amino acid sequences.</p>
</div>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h1>
<p>The following is the session info that generated this vignette:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## R Under development (unstable) (2020-10-24 r79367)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] grid      stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] seqLogo_1.57.0   BiocStyle_2.19.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] knitr_1.30          magrittr_1.5        R6_2.5.0           </span>
<span class="co">##  [4] ragg_0.4.0          rlang_0.4.8         highr_0.8          </span>
<span class="co">##  [7] stringr_1.4.0       tools_4.1.0         xfun_0.19          </span>
<span class="co">## [10] htmltools_0.5.0     systemfonts_0.3.2   yaml_2.2.1         </span>
<span class="co">## [13] assertthat_0.2.1    rprojroot_1.3-2     digest_0.6.27      </span>
<span class="co">## [16] pkgdown_1.6.1.9000  crayon_1.3.4        bookdown_0.21      </span>
<span class="co">## [19] textshaping_0.1.2   BiocManager_1.30.10 fs_1.5.0           </span>
<span class="co">## [22] memoise_1.1.0       evaluate_0.14       rmarkdown_2.5      </span>
<span class="co">## [25] stringi_1.5.3       compiler_4.1.0      desc_1.2.0         </span>
<span class="co">## [28] backports_1.2.0     stats4_4.1.0</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Chambers1998">
<p>Chambers, John M. 1998. <em>Programming with Data: A Guide to the S Language</em>. Springer.</p>
</div>
<div id="ref-Schneider1990">
<p>Schneider, Thomas D., and R.Michael Stephens. 1990. “Sequence logos: a new way to display consensus sequences.” <em>Nucleic Acids Research</em> 18 (20): 6097–6100. <a href="https://doi.org/10.1093/nar/18.20.6097">https://doi.org/10.1093/nar/18.20.6097</a>.</p>
</div>
<div id="ref-Schneider1986">
<p>Schneider, Thomas D., Gary D. Stormo, Larry Gold, and Andrzej Ehrenfeucht. 1986. “Information Content of Binding Sites on Nucleotide Sequences.” <em>Journal of Molecular Biology</em> 188 (3): 415–31. <a href="https://doi.org/https://doi.org/10.1016/0022-2836(86)90165-8">https://doi.org/https://doi.org/10.1016/0022-2836(86)90165-8</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Oliver Bembom, Robert Ivanek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
